packages/rxsocketio/src/operators/conduit.test.js

```diff
@@ -69,8 +69,28 @@ describe('conduit', () => {
       3: [{connected: false}, actions.disconnect({reason: 'bad connection'})],
       4: [{connected: true}, actions.connect({client: {}})],
     });
-    const actual$ = createMessageBuffer(message$, ioEvent$);
+    const actual$ = message$.pipe(createMessageBuffer(ioEvent$));
     const expected$ = m.cold('-0------(23)|', messages);
     m.expect(actual$).toBeObservable(expected$);
   }));
+
+  it('should passthrough items only when the socket is ready', marbles(m => {
+    const messages = {
+      0: {body: 'yar'},
+      1: {body: 'arr'},
+      2: {body: 'yo'},
+      3: {body: 'ho'},
+    };
+    const message$ = m.cold('0--1--23----|', messages);
+    const ioEvent$ = m.cold('012--3--4|', {
+      0: [{connected: false}, actions.createClient({client: {}})],
+      1: [{connected: true}, actions.connect({client: {}})],
+      2: [{connected: true}, actions.newMessage({topic: 'message', message: {greeting: 'sup dawg'}})],
+      3: [{connected: false}, actions.disconnect({reason: 'bad connection'})],
+      4: [{connected: true}, actions.connect({client: {}})],
+    });
+    const actual$ = message$.pipe(passthroughMessageBuffer(ioEvent$));
+    const expected$ = m.cold('---1--------|', messages);
+    m.expect(actual$).toBeObservable(expected$);
+  }));
```

Comment:
This is a good example of where marble testing is actually useful...  I wanted to make sure that my buffering logic worked correctly.  This test is basically demonstrating that the conduit will actually emit messages when the web socket is operational (connected).